tests/Benchmarks/System.Buffers/Reader_Binary.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using System.Buffers.Reader;
+
+namespace System.Buffers.Benchmarks
+{
+    public class Reader_Binary
+    {
+        static byte[] s_array;
+        static ReadOnlySequence<byte> s_ros;
+        static byte[] s_buffer;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            s_array = new byte[100000];
+            s_buffer = new byte[4096];
+            Random r = new Random(42);
+            r.NextBytes(s_array);
+            s_ros = new ReadOnlySequence<byte>(s_array);
+        }
+
+        [Benchmark]
+        public void ReadInt32()
+        {
+            var reader = BufferReader.Create(s_ros);
+
+            while (reader.TryRead(out int value))
+            {
+            }
+        }
+
+        [Benchmark]
+        public void PeekSpan()
+        {
+            const int Count = 1000;
+            const int Iterations = 100000;
```

Comment:
Why are we manually setting up the iterations here? Is the objective here to measure reader.Peek and amortize the overhead of the setup of Span/BufferReader (which are ref structs and can't be saved as fields in global setup)?

cc @adamsitnik

Reply:
@ahsonkhan yes, you are right. However this benchmark was more "ported" from xunit than written in "BDN style"

Reply:
Yes, I couldn't hold the reader as you mention.

Reply:
@adamsitnik, do you plan to add support for ref struct fields in BDN?

Reply:
@ahsonkhan no, at least as of today. It would imply too many changes for our engine (today we derive a class from type with benchmarks, with structs it's impossible) +the stack only has too many limitations, example: it can't be used for `[MemberData]` because it requires to cast to object

```cs
 public IEnumerator<object[]> GetEnumerator()
    {
        yield return new object[] { 1, 2, 3 };
        yield return new object[] { -4, -6, -10 };
        yield return new object[] { -2, 2, 0 };
        yield return new object[] { int.MinValue, -1, int.MaxValue };
    }
```

we also can't use `Func<Span<T>>` because it can't be generic argument ;(